src/toxcore/net_crypto/mod.rs

```
@@ -253,75 +249,60 @@ impl NetCrypto {
-            Err(e) => return Box::new(future::err(Error::from(e))),
+            Err(e) => return Box::new(future::err(HandlePacketError::from(e))),
         };
 
         if packet.cookie.hash() != payload.cookie_hash {
-            return Box::new(future::err(Error::new(
-                ErrorKind::Other,
-                "Invalid SHA512 hash of cookie"
-            )))
+            return Box::new(future::err(HandlePacketError::from(HandlePacketErrorKind::Sha512)))
         }
 
         let cookie = match packet.cookie.get_payload(&self.symmetric_key) {
             Ok(cookie) => cookie,
-            Err(e) => return Box::new(future::err(Error::from(e))),
+            Err(e) => return Box::new(future::err(HandlePacketError::from(e))),
         };
 
         if cookie.is_timed_out() {
-            return Box::new(future::err(Error::new(
-                ErrorKind::Other,
-                "Cookie is timed out"
-            )))
+            return Box::new(future::err(HandlePacketError::from(
+                HandlePacketErrorKind::CookieTimedOut)))
         }
         if cookie.real_pk != connection.peer_real_pk {
-            return Box::new(future::err(Error::new(
-                ErrorKind::Other,
-                "Cookie contains invalid real pk"
-            )))
+            return Box::new(future::err(HandlePacketError::from(HandlePacketErrorKind::InvalidRealPk)))
         }
         if cookie.dht_pk != connection.peer_dht_pk {
             return Box::new(
                 send_to(&self.dht_pk_tx, (connection.peer_real_pk, cookie.dht_pk))
-                    .map_err(|e|
-                        Error::new(ErrorKind::Other,
-                            format!("Failed to send packet: {:?}", e)
-                    ))
-                    .and_then(|()| future::err(Error::new(
-                        ErrorKind::Other,
-                        "Cookie contains invalid dht pk"
-                    )))
+                    .map_err(|_|
```

Why this error is ignored?

---

done